src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -25,76 +24,69 @@ export type CheckboxInputProps = CheckboxHTMLElementProps & {
   id: string;
 };
 
-const checkboxIconNameMap = {
-  indeterminate: 'checkbox-indeterminate',
-  true: 'checkbox-checked',
-  false: 'checkbox-unchecked',
-};
-const CheckboxSvg = ({
-  checked,
-  disabled,
-}: {
-  checked: boolean | 'indeterminate';
-  disabled?: boolean;
-}) => {
-  const iconClassName = clsx(
-    styles['checkbox__icon'],
-    disabled && styles['checkbox__icon--disabled'],
-  );
-  return (
-    <Icon
-      className={iconClassName}
-      name={
-        checkboxIconNameMap[`${checked}`] as
-          | 'checkbox-indeterminate'
-          | 'checkbox-checked'
-          | 'checkbox-unchecked'
-      }
-      purpose="decorative"
-      size="1.5rem"
-    />
-  );
-};
-
 /**
  * Checkbox input element, exported for greater flexibility.
  * You must provide an `id` prop and connect it to a visible label.
  */
 export const CheckboxInput = React.forwardRef<
   HTMLInputElement,
   CheckboxInputProps
->(({ checked = false, className, disabled, ...other }, ref) => {
+>(({ checked, className, disabled, ...other }, ref) => {
   // Make indeterminate checkbox visually match the colors of a
   // checked state, but announce itself as "mixed" to screen readers
   const checkedProps =
     checked === 'indeterminate'
       ? {
           'aria-checked': 'mixed' as const,
-          checked: true,
+          // Indeterminate checkboxes are unchecked when submitted to the server.
+          checked: false,
```

Comment:
Fixed a logical bug here.

We were counting "indeterminate" checkboxes as "checked", but that's not right. They're supposed to be functionally the same as unchecked.

Reply:
TODO: add a test for this behavior.

Reply:
Reverted this change, even though it's correct. I want to handle this in another commit/PR.